sdk/queue/priority_queue.go

```diff
@@ -85,13 +85,13 @@ func (pq *PriorityQueue) Len() int {
 // wrapper/convenience method that calls heap.Pop, so consumers do not need to
 // invoke heap functions directly
 func (pq *PriorityQueue) Pop() (*Item, error) {
+	pq.lock.Lock()
+	defer pq.lock.Unlock()
+
 	if pq.Len() == 0 {
```

Comment:
We'd want to change this to call `pq.data.Len()` directly to avoid grabbing a read lock and deadlocking.

Reply:
Done. Good catch!